### PR TITLE
fix #8817 chore(cirrus): only copy cirrus files into cirrus container

### LINKED ISSUE #8817: Only copy cirrus files into cirrus container
Right now we're copying the entire Experimenter repo into the cirrus container but we only need to copy the cirrus files into its container.  Let's qualify the paths and update the commands.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3480)

### PR DESCRIPTION
**Pull request by jaredlockhart**
Becuase
    
* We were copying the entire experimenter repo into the cirrus container
* The cirrus container only needs the cirrus files
    
This commit
    
* Only copies the cirrus files into the cirrus container
* Removes unnecessary top level python module files and updates import paths
* Updates all path references
* Build cirrus container on every cirrus make command
* Remove duplicate cirrus entry from test compose file

### CODE REVIEW

**Comment by jaredlockhart** [comment]
Blocked on https://github.com/mozilla/experimenter/pull/8813

File: cirrus/server/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+**/__pycache__
+**/.cache
+**/.coverage
+**/.DS_Store
+**/.mypy
```
[the remark this answers is not in the document]
Comment (jaredlockhart) [reply cirrus/server/.dockerignore:5]: Oh I just copied from the experimenter one, no reason not to be exhaustive.

**Review by yashikakhurana — APPROVED** [review]
Tested all commands locally, and everything seems perfect 🌷 \o/

File: Makefile [hunk 2]
```diff
@@ -208,34 +208,39 @@ integration_test_nimbus_rust: build_prod
...
-CIRRUS_BLACK_FIX = black -l 90 cirrus/server
-CIRRUS_RUFF_CHECK = ruff cirrus/server
-CIRRUS_RUFF_FIX = ruff --fix cirrus/server
-CIRRUS_PYTEST = pytest cirrus/server --cov-config=cirrus/server/.coveragerc --cov=cirrus/server/cirrus
-CIRRUS_PYTHON_TYPECHECK = pyright -p cirrus/server
-CIRRUS_PYTHON_TYPECHECK_CREATESTUB = pyright -p cirrus/server --createstub cirrus
-CIRRUS_GENERATE_DOCS = python cirrus/server/cirrus/generate_docs.py
-
-cirrus_up:
+CIRRUS_BLACK_CHECK = black -l 90 --check --diff .
```
[the remark this answers is not in the document]
Comment (jaredlockhart) [reply Makefile:211]: Yeah so now `/app/cirrus` contains the contents of `experimenter/cirrus/server/cirrus` so the python project is at the root

File: docker-compose.yml [hunk 3]
```diff
@@ -104,14 +104,14 @@ services:
 
   cirrus:
     build:
-      context: .
-      dockerfile: cirrus/server/Dockerfile
+      context: cirrus/server/
```
[the remark this answers is not in the document]
Comment (jaredlockhart) [reply docker-compose.yml:107]: Yeah exactly, so only the contents of `cirrus/server` are included instead of the entire experimenter repo 👍

**Pull request merged**